src/controller/CHIPDeviceController.cpp

```diff
@@ -2238,247 +2363,228 @@ void DeviceCommissioner::ParseCommissioningInfo()
-                        {
-                            info.network.thread.endpoint = path.mEndpointId;
-                        }
-                        if (info.network.wifi.endpoint == kInvalidEndpointId)
-                        {
-                            info.network.wifi.endpoint = path.mEndpointId;
-                        }
-                    }
-                }
-            }
-            return CHIP_NO_ERROR;
-        });
-    return_err = err == CHIP_NO_ERROR ? return_err : err;
-
-    err = mAttributeCache->ForEachAttribute(
-        app::Clusters::NetworkCommissioning::Id, [this, &info](const app::ConcreteAttributePath & path) {
-            using namespace chip::app::Clusters::NetworkCommissioning::Attributes;
-            if (path.mAttributeId != ConnectMaxTimeSeconds::Id)
+    err = mAttributeCache->ForEachAttribute(NetworkCommissioning::Id, [this, &info](const ConcreteAttributePath & path) {
+        VerifyOrReturnError(path.mAttributeId == FeatureMap::Id, CHIP_NO_ERROR);
+        BitFlags<NetworkCommissioning::Feature> features;
+        if (mAttributeCache->Get<FeatureMap::TypeInfo>(path, *features.RawStorage()) == CHIP_NO_ERROR)
+        {
+            if (features.Has(NetworkCommissioning::Feature::kWiFiNetworkInterface))
             {
-                return CHIP_NO_ERROR;
+                ChipLogProgress(Controller, "NetworkCommissioning Features: has WiFi. endpointid = %u", path.mEndpointId);
+                info.network.wifi.endpoint = path.mEndpointId;
             }
-            ConnectMaxTimeSeconds::TypeInfo::DecodableArgType time;
-            ReturnErrorOnFailure(this->mAttributeCache->Get<ConnectMaxTimeSeconds::TypeInfo>(path, time));
-            if (path.mEndpointId == info.network.wifi.endpoint)
+            else if (features.Has(NetworkCommissioning::Feature::kThreadNetworkInterface))
             {
-                info.network.wifi.minConnectionTime = time;
+                ChipLogProgress(Controller, "NetworkCommissioning Features: has Thread. endpointid = %u", path.mEndpointId);
+                info.network.thread.endpoint = path.mEndpointId;
             }
-            else if (path.mEndpointId == info.network.thread.endpoint)
+            else if (features.Has(NetworkCommissioning::Feature::kEthernetNetworkInterface))
             {
-                info.network.thread.minConnectionTime = time;
+                ChipLogProgress(Controller, "NetworkCommissioning Features: has Ethernet. endpointid = %u", path.mEndpointId);
+                info.network.eth.endpoint = path.mEndpointId;
             }
-            else if (path.mEndpointId == info.network.eth.endpoint)
+            else
             {
-                info.network.eth.minConnectionTime = time;
+                ChipLogProgress(Controller, "NetworkCommissioning Features: no features.");
+                // TODO: Gross workaround for the empty feature map on all clusters. Remove.
```

Comment:
That is a huge bug in all clusters. The right thing to do is to fix it.

Reply:
Yeah, I'll have a look if this is still needed (the workaround is existing code).

Reply:
I'm not sure this even happens any more - do you know if you're actually getting a blank feature map back?

Reply:
Yeah the current version of network-commissioning-server never returns a 0 feature map, I'll remove this.